cli/command/service/progress/progress.go

```diff
@@ -303,7 +327,23 @@ func (u *replicatedProgressUpdater) update(service swarm.Service, tasks []swarm.
 			u.slotMap[task.Slot] = mappedSlot
 		}
 
-		if !u.done && replicas <= maxProgressBars && uint64(mappedSlot) <= replicas {
+		if !terminalState(task.DesiredState) && task.Status.State == swarm.TaskStateRunning {
+			running++
+		}
+
+		if u.done || replicas > maxProgressBars || uint64(mappedSlot) > replicas {
```

Comment:
> I guess the purpose of this check is to not display progress bars if there are too many replicas? Couldn't this be handled at the top of the function?

We also have to count the number of running tasks, which this loop does as well.